external-crates/move/documentation/book/src/enums.md

```diff
@@ -0,0 +1,337 @@
+    //                                                     ^ ERROR! duplicate ability declaration
+}
+```
+
+For more details, see the section on
+[annotating abilities](./abilities.md#annotating-structs-and-enums).
+
+## Naming
+
+Enums and variants within enums must start with a capital letter `A` to `Z`. After the first letter,
+enum names can contain underscores `_`, lowercase letters `a` to `z`, uppercase letters `A` to `Z`,
+or digits `0` to `9`.
+
+```move
+public enum Foo { Variant }
+public enum BAR { Variant }
+public enum B_a_z_4_2 { V_a_riant_0 }
+```
+
+This naming restriction of starting with `A` to `Z` is in place to give room for future language
+features.
+
+## Using Enums
+
+### Creating Enum Variants
+
+Values of an enum type can be created (or "packed") by indicating a variant of the enum, followed by
+a value for each field in the variant. The variant name must always be qualified by the enum's name.
+
+Similarly to structs, for a variant with named fields, the order of the fields does not matter but
+the field names need to be provided. For a variant with positional fields, the order of the fields
+matters and the order of the fields must match the order in the variant declaration. It must also be
+created using `()` instead of `{}`. If the variant has no fields, the variant name is sufficient and
+no `()` or `{}` needs to be used.
+
+```move
+module a::m {
+    public enum Action has drop {
+        Stop,
+        Pause { duration: u32 },
+        MoveTo { x: u64, y: u64 },
+        Jump(u64),
+    }
+    public enum Other has drop {
+        Stop(u64),
+    }
+
+    fun example() {
+        // Note: The `Stop` variant of `Action` doesn't have fields so no parentheses or curlies are needed.
+        let stop = Action::Stop;
+        let pause = Action::Pause { duration: 10 };
+        let move_to = Action::MoveTo { x: 10, y: 20 };
+        let jump = Action::Jump(10);
+        // Note: The `Stop` variant of `Other` does have positional fields so we need to supply them.
+        let other_stop = Other::Stop(10);
+    }
+}
+```
+
+For variants with named fields you can also use the shorthand syntax that you might be familiar with
+from structs to create the variant:
+
+```move
+let duration = 10;
+
+let pause = Action::Pause { duration: duration };
+// is equivalent to
+let pause = Action::Pause { duration };
+```
+
+### Pattern Matching Enum Variants and Destructuring
+
+Since enum values can take on different shapes, dot access to fields of variants is not allowed like
+it is for struct fields. Instead, to access fields within a variant -- either by value, or immutable
+or mutable reference -- you must use pattern matching.
+
+You can pattern match on Move values by value, immutable reference, and mutable reference. When
+pattern matching by value, the value is moved into the match arm. When pattern matching by
+reference, the value is borrowed into the match arm (either immutably or mutably). We'll go through
```

Comment:
I thought we tended not to use "We" in docs. cc @ronny-mysten, did I remember that correctly, and if so, what's the alternative in cases like this?

Reply:
That's true for Sui docs. These don't have a style guide and was asked not to provide that level of feedback originally. It's better to just mention that "the instructions...", "the document...", "the following examples..." go through... But because "we" was used throughout, probably best to keep it consistent.